api-docs/client-files.md

```diff
@@ -0,0 +1,274 @@
+# Client files
+
+This api is provided to upload and keep track of client files.
+To upload or access any file you need the corresponding signed url.
+
+### API flow
+
+If performing an upload you will get a signed url via `/api/signed-file-uploads` you will then make
+a PUT request to the url like so
+
+```javascript
+axios.put(signedRequest, file, options).then(result => {
+  if (result.data.success) {
+    //Make the request to create a client file /api/clients/:clientId/file
+  }
+});
+```
+
+After you upload the file to S3, you need to tell the web server that you
+did so by calling `http POST /api/clients/:clientId/files`
+
+For each file that you want to download you would make a request to `/api/clients/:clientId/files/:fileId/signed-downloads` then
+you would make a request on the front end like so `window.location.href = downloadurl`
+
+## Get a signed POST URL
+
+```http
+GET /api/signed-file-uploads
```

Comment:
Since you and I talked about how the `fileKey` returned from the server is going to have the client directory and a uuid for filename, we should update this API so that it is specific to a client so that the server knows which directory to put the filekey in.

```suggestion
GET /api/clients/:clientId/signed-file-uploads
```